HCCGo/app/js/GenericClusterInterface.js

```diff
@@ -12,25 +12,102 @@ var GenericClusterInterface = function(connectionService, $q) {
 }
 
 GenericClusterInterface.prototype.getJobs = function() {
-  
+
   // return a promise if the jobs info are found
-  
+
+}
+
+function CSVToArray( strData, strDelimiter ){
+  strDelimiter = (strDelimiter || ",");
+  var objPattern = new RegExp(
+    (
+        // Delimiters.
+        "(\\" + strDelimiter + "|\\r?\\n|\\r|^)" +
+        // Quoted fields.
+        "(?:\"([^\"]*(?:\"\"[^\"]*)*)\"|" +
+        // Standard fields.
+        "([^\"\\" + strDelimiter + "\\r\\n]*))"
+    ),
+    "gi"
+  );
+  var arrData = [[]];
+  var arrMatches = null;
+  while (arrMatches = objPattern.exec( strData )){
+    var strMatchedDelimiter = arrMatches[ 1 ];
+    if (
+        strMatchedDelimiter.length &&
+        strMatchedDelimiter !== strDelimiter
+        ){
+        arrData.push( [] );
+    }
+    var strMatchedValue;
+    if (arrMatches[ 2 ]){
+        strMatchedValue = arrMatches[ 2 ].replace(
+            new RegExp( "\"\"", "g" ),
+            "\""
+            );
+
+    } else {
+        strMatchedValue = arrMatches[ 3 ];
+    }
+    arrData[ arrData.length - 1 ].push( strMatchedValue );
+  }
+  return( arrData );
 }
 
+GenericClusterInterface.prototype.getCompletedJobs = function(docs) {
+  var deferred = this.$q.defer();
+  var completedJobs = [];
+  for(var i = 0; i < docs.length; i++) {
+    // setup return document
+    var returnData = {
+      "_id": docs[i]._id
+    }
+    this.connectionService.runCommand("sacct -j " + docs[i].jobId + " -l --noconvert -P").then(function(data) {
```

Comment:
This should probably be in the Slurm specific interface, rather than the generic cluster interface.